### PR TITLE
chore: Convert metric tags tests to use metric_tags! macro

### PR DESCRIPTION
**Pull request by bruceg**
Found a bunch of these metric tags that could be simplified with the macro introduced in #14835 while dealing with type issues in the follow-up to #14984. These should all have no behavior changes.

### CODE REVIEW

File: src/sources/host_metrics/network.rs [hunk 1]
```diff
@@ -44,42 +44,43 @@ impl HostMetrics {
...
                     );
                     #[cfg(any(target_os = "linux", windows))]
                     output.counter(
                         "network_transmit_packets_total",
                         counter.packets_sent() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
+                    );
+                    output.counter(
+                        "network_transmit_errs_total",
```
Comment (spencergilbert) [inline src/sources/host_metrics/network.rs:81]: Curious why you moved this to the bottom? It was alphabetical, and the `recv` side still is.

**Review by spencergilbert — APPROVED** [review]
One question about a reordering, but not a blocker.

File: src/sources/host_metrics/network.rs (hunk 1 above)
Comment (bruceg) [reply src/sources/host_metrics/network.rs:81]: That lets the last `tags` avoid the `.clone()`. The last use of value cannot be cloned due to a clippy check, and the last two metrics are optional without reordering, causing issues.

Comment (spencergilbert) [reply src/sources/host_metrics/network.rs:81]: Ah, 👍 I missed that and just thought it was an arbitrary reordering - like errors last or something 😄

**Pull request merged**